pkg/route/validation/validation.go

```diff
@@ -90,6 +99,33 @@ func validateRoute(route *routev1.Route, checkHostname bool) field.ErrorList {
 		result = append(result, err)
 	}
 
+	if len(route.Spec.HTTPHeaders.Actions.Response) != 0 || len(route.Spec.HTTPHeaders.Actions.Request) != 0 {
+		if route.Spec.TLS != nil && route.Spec.TLS.Termination == routev1.TLSTerminationPassthrough {
+			specTLS := field.NewPath(specTLSTerminationString)
+			result = append(result, field.NotSupported(specTLS, route.Spec.TLS.Termination, []string{"Only edge and re-encrypt routes are supported for providing customized headers."}))
+		}
+	}
+	if len(route.Spec.HTTPHeaders.Actions.Response) != 0 && len(route.Spec.HTTPHeaders.Actions.Response) > 64 {
+		result = append(result, field.Invalid(specPath.Child(specResponseString), route.Spec.HTTPHeaders.Actions.Response, "http response headers list can't exceed more than 64 items."))
+	}
+
+	if len(route.Spec.HTTPHeaders.Actions.Request) != 0 && len(route.Spec.HTTPHeaders.Actions.Request) > 64 {
+		result = append(result, field.Invalid(specPath.Child(specRequestString), route.Spec.HTTPHeaders.Actions.Request, "http request headers list can't exceed more than 64 items."))
```

Comment:
We don't have any unit test coverage for this case.

Reply:
added